examples/all-clusters-app/all-clusters-common/src/WhmManufacturer.cpp

```diff
@@ -0,0 +1,270 @@
+
+    uint8_t heaterDemandMask = 0;
+    for (uint8_t idx = 0; idx < sizeof(waterHeaterTypeValues) / sizeof(waterHeaterTypeValues[0]); idx++)
+    {
+        // Is this heating source being used?
+        if (heaterTypes.Raw() & waterHeaterTypeValues[idx])
+        {
+            heaterDemandMask |= waterHeaterDemandValues[idx];
+        }
+    }
+
+    return BitMask<WaterHeaterDemandBitmap>(heaterDemandMask);
+}
+
+Status WhmManufacturer::TurnHeatingOn()
+{
+    Status status = Status::Success;
+
+    ChipLogProgress(AppServer, "WhmManufacturer::TurnHeatingOn");
+
+    WaterHeaterManagementDelegate * dg = GetWhmDelegate();
+
+    if (dg->GetBoostState() == BoostStateEnum::kActive)
+    {
+        mBoostActive = true;
+    }
+
+    return status;
+}
+
+Status WhmManufacturer::TurnHeatingOff()
+{
+    Status status = Status::Success;
+
+    ChipLogProgress(AppServer, "WhmManufacturer::TurnHeatingOff");
+
+    if (mBoostActive)
+    {
+        mBoostActive = false;
+    }
+
+    return status;
+}
+
+Status WhmManufacturer::BoostCommandStarted(uint32_t duration, Optional<bool> oneShot, Optional<bool> emergencyBoost,
+                                            Optional<int16_t> temporarySetpoint, Optional<chip::Percent> targetPercentage,
+                                            Optional<chip::Percent> targetReheat)
+{
+    return Status::Success;
+}
+
+Status WhmManufacturer::BoostCommandCancelled()
+{
+    return Status::Success;
+}
+
+void WhmManufacturer::BoostCommandFinished() {}
+
+WaterHeaterManagementDelegate * GetWhmDelegate()
+{
+    WhmManufacturer * mn = GetWhmManufacturer();
+    VerifyOrDieWithMsg(mn != nullptr, AppServer, "WhmManufacturer is null");
+
+    WaterHeaterManagementDelegate * wg = mn->GetWhmDelegate();
+    VerifyOrDieWithMsg(wg != nullptr, AppServer, "WhmDelegate is null");
+
+    return wg;
+}
+
+void SetTestEventTrigger_BasicInstallationTestEvent()
+{
+    WaterHeaterManagementDelegate * dg = GetWhmDelegate();
+
+    // Simulate installation in a 100L tank full of water at 20C, with a target temperature of 60C, in OFF mode
+    dg->SetTankVolume(100);
+    dg->SetTargetWaterTemperature(6000);
+    dg->SetHeaterTypes(BitMask<WaterHeaterTypeBitmap>(WaterHeaterTypeBitmap::kImmersionElement1));
+    dg->DrawOffHotWater(100, 2000);
+}
+
+void SetTestEventTrigger_BasicInstallationTestEventClear() {}
+
+void SetTestEventTrigger_WaterTemperature20CTestEvent()
+{
+    WaterHeaterManagementDelegate * dg = GetWhmDelegate();
+
+    // Simulate 100% of the water in the tank being at 20C
+    dg->SetWaterTemperature(2000);
+}
+
+void SetTestEventTrigger_WaterTemperature61CTestEvent()
+{
+    WaterHeaterManagementDelegate * dg = GetWhmDelegate();
+
+    // Simulate 100% of the water in the tank being at 61C
+    dg->SetWaterTemperature(6100);
+}
+
+void SetTestEventTrigger_WaterTemperature66CTestEvent()
+{
+    WaterHeaterManagementDelegate * dg = GetWhmDelegate();
+
+    // Simulate 100% of the water in the tank being at 66C
```

Comment:
As above the '100%' just adds confusion

Reply:
That's the exact text describing the trigger in the spec. Does that mean the spec needs to rephrased?

Reply:
We can do that in a follow up PR when the spec 10028 has merged.